### ISSUE TITLE
[18.0] web_responsive and main page apps

### ISSUE THREAD

**Issue by bealdav** [OCA/web#2984]
Thanks for your fantastic work on responsive module

Just a question.

## main page app
![image](https://github.com/user-attachments/assets/af18b409-8398-49e3-b3ba-3daa8fcf7522)

## community drop down app list
![image](https://github.com/user-attachments/assets/526f67f4-b408-486a-bc9a-018a7f9d14f4)

**Describe the solution you'd like**
Related to the migration of web_responsive in 18.0.
This module is complex to migrate because of many features.

When a new version arrives it could be good to have a `main app page` earlier in the migration process to demonstrate to new customers that community version is nice.

Then is it possible to extract the `main page app` with nice icons from web_responsive module, because you may want to have only this main page apps. Months later you might want responsive features without this main page app with only `community drop down app` according to your choice.

The drop down is fine to directly select your menu in a productive way, while the main page let  your eyes  spread anywhere on the page to find the right app.
Having the choices is good.

I probably forgot contributors here, please alert them

cc @yajo  @legalsylvain @etobella @Tardo @angelmoya @chienandalu @SplashS

**Comment by legalsylvain** [comment]
Hi. context : I never worked on web_responsive and I have limited skills in Javascript / CSS.

In fact, I don't understand in which case we want the community drop down list. 
- I see it very poor
- there is no possibility to search, so it requires more click
- I don't see it very beautiful. (matter of taste on that topic).

+ maintain 2 (or more) modules is more work in each migration.

So I'm in favor to keep this module as it. 

Basically this module takes part of the "must have" modules and tools that should be migrated each year by the OCA. I prefer to work on the topic of how the OCA sustains financialy that work (cf talk in the OCA days).

My two cents.

**Comment by PaulGoubert** [comment]
I agree with David’s vision and understand Sylvain’s point: web_responsive is a must-have.

From my perspective, the web_responsive module includes the main menu with icons, which is nice and sexy for new users. It mirrors the way they use apps on their phones.

When a new version is released, it makes sense to base new projects on that version and enhance the UI with this more visual menu, thanks to the icons.

The web_responsive module is comprehensive and includes all the responsiveness features needed for use on tablets, for instance. However, this isn’t necessarily critical in the early months of a new project.

On the other hand, the web interface for the homepage menu allows us to present a more user-friendly side of the ERP during new user training sessions. It feels less intimidating than the dropdown menu of the CE version.

Having a module like web_app_menu or something similar that focuses solely on this feature would be a great addition. It might also be easier to migrate between versions while waiting for the full web_responsive module, with all its features, to become available in the months following a new release.

I have a client who might be interested and potentially willing to partially fund this development.

Again, this is purely cosmetic, but it makes sense from a marketing perspective for our CE version.

**Comment by Tardo** [comment]
It's been a few years since I've been involved in this module.
I don't know in which version the side menu behavior was separated. I think the app menu feature is a good candidate to be separated as well. It goes beyond “responsive” design... It offers a functionality
I think that a module with concrete functions helps to get other developers involved. Working with a “big” module always gives more 'laziness' to get involved. At least that's how I see it.

**Comment by chienandalu** [comment]
I might agree with the splitting idea :smile:

**Comment by pedrobaeza** [comment]
I prefer to have it together. At the end, we will always install both. Which case do you need one and not the other?

**Comment by houssine78** [comment]
I also agree having it spitted. But mostly to have the icons menu available quickly.

My two cents on this topic.

**Comment by yajo** [comment]
FWIW, if you hit <kbd>Ctrl</kbd>+<kbd>K</kbd>  and type <kbd>/</kbd> + the menu you want, you can search for it without this module in Odoo 16 CE. Not tested in 18.

**Comment by bealdav** [comment]
Thanks @yajo for the clarification.

**Comment by bealdav** [comment]
@pedrobaeza The topic @PaulGoubert raises is not his/our preferences, it only concerns prospects' preferences. 
When a new version of Odoo arrives, @PaulGoubert tries to convince new potential customers to bet on Odoo's Community OCA ecosystem. 
He is frequently in front of other partners who sell and demonstrate the Odoo enterprise with a nice landing page. This may be a detail for you, but for these people (prospects), such a landing page can be decisive for their choice. 

We are committed to offering the latest version of Odoo when it is released. Then, @PaulGoubert proposes a solution to make the OCA offer attractive a few weeks after the release of a new version. And he offers to contribute financially to an OCA expert to make such a module easier and faster to migrate. 

I see that people's reactions tend to be positive to his proposal. 

And yet, reducing the scope of a module is good for quality and makes migration easier.

[EDIT] ... and make subset features migration earlier .

**Comment by pedrobaeza** [comment]
As said by @yajo: you don't need menu apps for searching menus and having a graphical interface. Just use command pallette with the `/` character. The menu app is part of the responsive experience for having a touching type display, so it shouldn't be dettached IMO.

**Comment by pedrobaeza** [comment]
> And yet, reducing the scope of a module is good for quality and makes migration easier.

No, this is not true a lot of times. Decoupling things require to create interfaces, hooks, etc for extensions to attach there, and the maintenance burden for keeping everything compatible is sometimes even more (and it always requires more lines of code).

**Comment by legalsylvain** [comment]
@SplashS : could you provide your point of view on this topic ? you are the second developper marked as "maintainers" in the latest released version. https://github.com/OCA/web/tree/17.0/web_responsive#maintainers
(@Tardo already gave an opinion here : https://github.com/OCA/web/issues/2984#issuecomment-2502054158)
Thanks !

**Comment by aradradev** [comment]
I tried the web_responsive by adding it to my dokcer-composer file, however, the version does not match. What I do have is version 16, and my Odoo is in version 18, that is why it did not work. How can I fix this? I'm sorry, I do not know where to ask this, because I am new to Odoo development, and I looked for an answer in many places like StackOverflow, but did not get an answer.
